src/lib/app-authenticator.js

```diff
@@ -1,3 +1,4 @@
+import AuthData from '../../node_modules/alpheios-components/src/lib/auth/auth-data.js'
```

Comment:
this seems a little at odds with our dynamic import approach for components
not sure if it's critical since we aren't using this module in production but just wondering about it

Reply:
I've faced an interesting problem and that's the best solution I was able to find. I would really love to hear your opinions on that.

`AuthData` is defined in `components`. Those are imported dynamically inside `embedded.js`. The only way to use `AuthData` in `AppAuthenticator` would be to pass it dynamically from `embedded.js` into the `AppAuthenticator` constructor. But that, on my opinion, would be overly complex.

Some other things we could do (I think all those solutions are worse than the one used):
1. Duplicate `AuthData` into the `embed-lib`.
2. Import `AuthData` from the `components`. Webpack will probably tree shake the rest of the classes, but the resulting code will probably still be larger and compilation will be way slower.
3. Move app and session authenticators into the `components`.

Are there any other solutions to the problem? Would would be the best way to solve the issue, on your opinion? This import from the dependency's source looks unusual, but is effective and efficient and I was not able to find any reasonable arguments against it (except that it's not pure enough and does rely on the knowledge of internal directory structure of the components library). What do you think?

P.S. Just had and idea that maybe a cleaner solution would be to export `auth-data.js` into the `dist` directory of `components` and import it from there. I've seen a couple of libraries doing that. But I'm not sure if it's worth the trouble. What do you think?

Reply:
I think the right approach in the long term is probably to move the authenticators into components.
But I don't think we need to do that now. AppAuthenticator is only used for demo purposes right now because we don't use the SPA approach. So as long as we won't get any weird errors when the code is imported the way it is, I'm fine with it.

Reply:
I agree with Bridget!
I tink that may be it is worth to separate all authentication code in a separate private library for security reasons.